### ISSUE TITLE
Using Binary?

### ISSUE THREAD

**Issue by farberm** [kk7ds/pynx584#41]
Is there a simple way to convert this to use Binary instead of ASCII? (My guess is that the answer is no) My NX-8E is already setup to use Binary and I would have to turn off my old system to make the switch over to ASCII from Binary communication.
 Thanks in advance

**Comment by heatvent** [comment]
Same here, except my panel is locked by the installer (I don't have the program code).  Any chance this could support binary communication?

**Comment by kk7ds** [comment]
Binary isn't supported as of now, and I don't have any need or desire to implement it. But, someone else certainly could. I think it's important to note that it's quite likely that your installer didn't enable enough of the external communication port to be really useful anyway. FWIW, I called the original installer and got them to unlock the panel for me.

**Comment by heatvent** [comment]
Thanks for the reply. Mine told me it was against their policy to give the
installer code.  The panel was setup for communication with a Control4 HA
system.  Not sure of the exact settings and exactly what was turned on or
off.  Just know it is binary.

**Comment by kk7ds** [comment]
Yeah, so did mine, so I told them that I'd be firing them right after they came and unlocked my panel :)

They literally use the same code for everyone they install, which is ridiculous "security" anyway. I watched the guy unlock my panel to set it to something else, which means I could reprogram anyone else's that use their service.

**Comment by Pre10der** [comment]
I had the same issue, and I opted to default my panel and reconfigure it from scratch. It was daunting at first, but I learned a lot from it...oh, and I became comfortable that nobody else new the master/installer code. This might be something to consider.

**Comment by heatvent** [comment]
So I am on that path ... somewhat. I got new a new old stock NX-8e board
and cabinet off of Ebay for I think under $30.  However, I really want to
see how the current one is programmed and use DL900 vs keypunching
everything.  So I also picked up a NX-588E for $30.  Not sure if they
changed the download code or not but really want to be able to download the
current settings if I can.

One thought I had at one point is I don't know if any of the extensions
need a pin code to access information. The Control4 did so you had to use
the Master Code to operate anything. I think the plugin could be modified
(if it were using binary) to cycle through the 10k combinations and stop
when it successfully connects with the correct PIN.  Almost wanted to do
this just out of spite.
